### PR TITLE
theconfluenceBot/new.py Harding of possible failure conditions. corrected some linter errors. Simplified get_random_posts

### PR DESCRIPTION
**Pull request by Thejusko**
random changed to rand to use different name from import library
removed unused import
Prevented use of undefined reddit object 
Simplified get_random_posts
Fixed some linter errors

The while true in main was removed. This was done because a while true is active waiting which indefinitly blocks the used CPU  Thread. Which the change to a stream the Script only uses CPU Time when a new message is received.

### CODE REVIEW

File: theconfluenceBOT/new.py [hunk 1]
```diff
@@ -114,6 +119,8 @@ def departures():
         print("Logged in as Zokalyx!")
     except Exception as e:
         print(e)
+        # Prevents the use of an undefined reddit object
+        sys.exit("Zokalyx login in departures failed.")
```
Comment (Zokalyx) [inline theconfluenceBOT/new.py]: Wouldn't this prevent the bot from replying with "there was a problem"?

File: theconfluenceBOT/new.py [hunk 2]
```diff
@@ -410,27 +420,27 @@ def get_random_posts(reddit):
...
-        return "popular", sub.new(limit=10)
-
     # 10% chance of r/all
     elif rand < 10:
         sub = reddit.subreddit("all")
-        return "all", sub.hot(limit=10)
+
+    if sub is not None:
+        hot_or_new = random.randint(0, 1)
+        # 50-50 chance of sorting by new or hot
```
Comment (Zokalyx) [inline theconfluenceBOT/new.py:484]: hot_or_new selection should be specific to random subreddit conditional branch (like it was before)
https://i.imgur.com/lWHrfAN.jpeg

**Review by Zokalyx — CHANGES_REQUESTED** [review]
Looks really good! But there are two things that have to be fixed or maybe I didn't understand the change...
By the way, sorry, I'm not very good with github, this is my first time using an actual function from the site apart from just having a repo.
I'm open to chat with you about the project and we can discuss changes together! Also, I can explain what each file does. Of course - no pressure nor obligation.

File: theconfluenceBOT/new.py (hunk 1 above)
Comment (Thejusko) [reply theconfluenceBOT/new.py]: new commit raises an Exception

File: theconfluenceBOT/new.py (hunk 2 above)
Comment (Thejusko) [reply theconfluenceBOT/new.py:484]: is changed back to conditional behaivior

File: theconfluenceBOT/new.py [hunk 3]
```diff
@@ -24,8 +24,7 @@
 WHITELIST = [
     "Zokalyx",
     "MoscaMye",
-    "theconfluencer",
-    "Thejusko"
```
Comment (Zokalyx) [inline theconfluenceBOT/new.py]: You can still add yourself - but I understand that you cannot change the DEBUG variable. Do you think it's a good idea to implement some sort of keyword that corresponds to debugging? Results would be sent back through DMs, but not posted on the profile. I'm not saying you should add that, I can do it but I just wanted to know your opinion.

EDIT: Maybe debugging should always happen unless the sender is MoscaMye or theconfluencer. And if we really want to, a keyword to make it NOT debug would be used for other people.

File: theconfluenceBOT/new.py [hunk 4]
```diff
@@ -1,13 +1,22 @@
...
 import math
+import logging
+from praw.models import MoreComments
+from praw.exceptions import RedditAPIException
 from datetime import datetime
 from os import environ as env
 
+LOG_FORMAT = "%(levelname)-8s |%(asctime)-25s| %(name)-45s | %(funcName)-25s at line %(lineno)-5s:%(message)s\n"
+logging.basicConfig(
+    filename="/dev/stdout", filemode="w", level=logging.INFO, format=LOG_FORMAT
```
Comment (Zokalyx) [inline theconfluenceBOT/new.py:15]: This is really nice, but this is too cluttered - seems like Heroku already provides some of the same logging format
image

File: theconfluenceBOT/new.py [hunk 5]
```diff
@@ -188,7 +239,7 @@ def get_departures(reddit):
             lastpop = len(lastrundata)
 
     # Check if runs are the same
-    if lastrun_sub != lastrun_bot:
+    if lastRunNumberSub != lastrun_bot:
```
Comment (Zokalyx) [inline theconfluenceBOT/new.py:242]: `lastrun_bot` is a string and `lastRunNumberSub` is an integer. Don't worry I'll change this (make `lastrun_bot` an int).

File: theconfluenceBOT/new.py (hunk 4 above)
Comment (Zokalyx) [reply theconfluenceBOT/new.py:15]: Also - I don't think this is supported on Windows (`/dev/stdout`). Not really an issue but might be good to add for local testing.

**Review by Zokalyx — APPROVED** [review]
This is great work! I will create issues of things to-do, but everything looks pretty nice. Thanks

**Pull request merged**